src/Features/Core/Portable/Extensions/IExtensionMessageHandlerFactory.cs

```diff
@@ -5,13 +5,14 @@
 using System.Collections.Immutable;
 using System.Reflection;
 using System.Threading;
+using Microsoft.CodeAnalysis.Host;
 
 namespace Microsoft.CodeAnalysis.Extensions;
 
 /// <summary>
 /// Factory for creating instances of extension message handlers.
 /// </summary>
-internal interface IExtensionMessageHandlerFactory
+internal interface IExtensionMessageHandlerFactory : IWorkspaceService
```

Comment:
became a workspace service so we can override it in tests with our own version ot mock out exceptions it might through without actually loading assemblies and using reflection on them.